harness/determined/searcher/search_runner.py

```diff
@@ -49,94 +34,189 @@ def run(
                         # duplicate completion accounting already performed by master
                         logging.info(
-                            f"validationCompleted({e.validationCompleted.requestId},"
-                            f" {e.validationCompleted.metric})"
+                            f"validationCompleted({event.validationCompleted.requestId},"
+                            f" {event.validationCompleted.metric})"
                         )
-                        request_id = uuid.UUID(e.validationCompleted.requestId)
-                        if e.validationCompleted.metric is None:
+                        request_id = uuid.UUID(event.validationCompleted.requestId)
+                        if event.validationCompleted.metric is None:
                             raise RuntimeError(
                                 "validationCompleted event is invalid without a metric"
                             )
                         operations = self.search_method.on_validation_completed(
                             request_id,
-                            e.validationCompleted.metric,
+                            event.validationCompleted.metric,
                         )
-                    elif e.experimentInactive:
+                    elif event.experimentInactive:
                         logging.info(
-                            f"experiment {experiment_id} is inactive"
-                            f" state={e.experimentInactive.experimentState}"
+                            f"experiment {self.search_method.searcher_state.experiment_id} is "
+                            f"inactive; state={event.experimentInactive.experimentState}"
                         )
                         experiment_is_active = False
                         break
-                    elif e.trialProgress:
+                    elif event.trialProgress:
                         logging.debug(
-                            f"trialProgress({e.trialProgress.requestId}, "
-                            f"{e.trialProgress.partialUnits})"
+                            f"trialProgress({event.trialProgress.requestId}, "
+                            f"{event.trialProgress.partialUnits})"
                         )
-                        request_id = uuid.UUID(e.trialProgress.requestId)
+                        request_id = uuid.UUID(event.trialProgress.requestId)
                         self.search_method.searcher_state.trial_progress[request_id] = float(
-                            e.trialProgress.partialUnits
+                            event.trialProgress.partialUnits
                         )
                         progress = self.search_method.progress()
                         operations = [Progress(progress)]
                     else:
-                        raise RuntimeError(f"Unsupported event {e}")
+                        raise RuntimeError(f"Unsupported event {event}")
                     bindings.post_PostSearcherOperations(
```

Comment:
i thought we'd check the last_event_id here and use that to decide whether we'd want to post operations or not.  or have some other way of skipping the first event if we already processed it client side.